metrics/network/cpu_statistics_iperf.sh

```diff
@@ -0,0 +1,98 @@
+#!/bin/bash
+#
+# Copyright (c) 2018 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# Description of the test:
+# This test runs the 'iperf3' network benchmark and
+# we retrieve cpu information while running the networking
+# test with 'perf'.
+
+set -e
+
+# General env
+SCRIPT_PATH=$(dirname "$(readlink -f "$0")")
+source "${SCRIPT_PATH}/../lib/common.bash"
+
+TEST_NAME="cpu information"
+
+# Image name
+image="networkstatic/iperf3"
+# Client name
+client_name="client_iperf"
+# Server name
+server_name="server_iperf"
+# Target bandwidth
+bandwidth="1G"
+# Time in seconds to transmit
+timeout="160"
+# Sample period in seconds to measure
+sample_period="120"
+# File to save perf results
+TMP_FILE=$(mktemp cpuinfo.XXXXXXXXXX)
+
+function main() {
+	# Check tools/commands dependencies
+	cmds=("awk" "docker" "perf")
+
+	init_env
+	check_cmds "${cmds[@]}"
+	check_images "$image"
+
+	# Start iperf3 server
+	local server_cmd="export TMPDIR=/dev/shm; iperf3 -s"
+	docker run --runtime $RUNTIME --name $server_name -d $image bash -c "$server_cmd"
+	local server_address=$(docker inspect --format "{{.NetworkSettings.IPAddress}}" $server_name)
+
+	# Start iperf3 client
+	local client_cmd="export TMPDIR=/dev/shm; iperf3 -c $server_address -b $bandwidth -t $timeout"
+	docker run --runtime $RUNTIME --name $client_name -d $image bash -c "$client_name"
+
+	# This time in seconds is required to let the benchmark settle down
+	sleep 20
+
+	# Retrieve qemu information
+	PIDS="$(pgrep -d ',' qemu)"
+
+	metrics_json_init
+
+	# Start collecting cpu information
+	sudo perf stat -o $TMP_FILE -e cycles -e instructions -p ${PIDS} sleep ${sample_period}
```

Comment:
if we are sat here waiting for this to complete (that is, we don't background it, which is fine), then maybe we can just capture the output directly, and skip the TMPFILE usage?